openqa-get-job-runtime-stats

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# 
+# Summary: Display the test runtime statistics for a range of openQA test runs
+# Job ranges can be defined via 7..9 (i,e, [7,8, 9]) or 5+2 (i.e. [5,6,7])
+# 
+# Usage example: ./openqa-get-job-runtime-stats https://openqa.opensuse.org/tests/100000+10
+# 
+
+import requests
+import datetime
+import time
+import json
+import sys
+import argparse
+import numpy as np
+
+def parse_t(dt_str) :
+	# "2023-01-25T10:22:21"
+	if dt_str is None : return None
+	dt_str = dt_str.strip()
+	if dt_str == "" : return None
+	return datetime.datetime.strptime(dt_str, '%Y-%m-%dT%H:%M:%S')
+
+
+class Job :
+	# Init job by the json object fetched from OpenQA
+	def __init__(self, obj) :
+		# Apply json values to this class. Yes, this is hacky.
+		for k in obj :
+			self.__dict__[k] = obj[k]
+	
+	def done(self) -> bool :
+		return self.state == "done"
+	def passed(self) -> bool :
+		return self.state == "done" and self.result == "passed"
+	def softfailed(self) -> bool :
+		return self.state == "done" and self.result == "softfailed"
+	def failed(self) -> bool :
+		return self.state == "done" and (self.result == "failed" or self.result == "incomplete")
```

Comment:
It is not good to call this `failed` when it is actually `failed` + `incomplete`. Note that there are also more states you might want to consider here. For instance, in openQA we have meta states/results, e.g. `OK_RESULTS` (see https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Jobs/Constants.pm#L76) and the negation is usually called `not_ok` (when used somewhere).

Reply:
Reverted the logic, please check again if you agree now.